tests/Unit/Adapter/Invoice/InvoicesConfigurationTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Invoice;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PrestaShop\PrestaShop\Adapter\Invoice\InvoiceOptionsConfiguration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Tests\TestCase\AbstractConfigurationTestCase;
+
+class InvoiceOptionsConfigurationTest extends AbstractConfigurationTestCase
+{
+    private const SHOP_ID = 42;
+
+    private const VALID_CONFIGURATION = [
+        'enable_invoices' => true,
```

Comment:
Seems like you are using the default configuration here, in the configuration class the field is defined as:

```php
'enable_invoices' => (bool) $this->configuration->get('PS_INVOICE', true, $shopConstraint),
```

meaning you can't really be sure that the value you are asserting is the one you are willing to test here or if it's the default one that is used as a fallback. To make sure everything is working fine it's safer to test values different from the default ones for all those fields.

I didn't check all of them, some are good some are not, but at least `invoice_prefix` also should be modified to a different value than the default one.

Reply:
Either that or you need to enrich your `provideValidConfiguration` function which different values for each field, but it's probably easier to handle it here.

Reply:
You are perfectly right, I fact I assumed partially this idea but I missed booleans ^^'

Thanks for good comments :)